build/OSX/CMakeLists.txt

```diff
@@ -1,210 +1,106 @@
-cmake_minimum_required(VERSION 3.15)
+cmake_minimum_required(VERSION 3.22)
 
-# --------------------------------------------------------
-# if on OSX Darwin system, build the OSX executables
-if (NOT ${CMAKE_HOST_SYSTEM_NAME} MATCHES "Darwin")
+if(NOT ${CMAKE_HOST_SYSTEM_NAME} MATCHES "Darwin")
     message(FATAL_ERROR "${CMAKE_HOST_SYSTEM_NAME} host system name does not match OSX Darwin - Please select the correct folder for configuring project")
 endif()
 
 if(POLICY CMP0025)
-	#NEW: report OS X version of Clang as "AppleClang"
-  cmake_policy(SET CMP0025 NEW)
+    cmake_policy(SET CMP0025 NEW)
 endif()
+
 if(POLICY CMP0042)
-  #NEW: set TARGET PROPERTY MACOSX_RPATH to on by default
-  cmake_policy(SET CMP0042 NEW)
+    cmake_policy(SET CMP0042 NEW)
 endif()
 
-set(CMAKE_OSX_DEPLOYMENT_TARGET "12.1" CACHE STRING "Minimum OS X deployment version" FORCE)
+if(NOT DEFINED CMAKE_TOOLCHAIN_FILE)
+    set(PLATFORM MAC CACHE STRING "Toolchain platform parameter")
+    set(CMAKE_TOOLCHAIN_FILE "${CMAKE_CURRENT_SOURCE_DIR}/../apple.toolchain.cmake" CACHE STRING "Toolchain file")
+endif()
 
-project(OSX_SHARED_THIRD_PARTY)
+project(OSX_SHARED_THIRD_PARTY LANGUAGES C CXX)
 
-# --------------------------------------------------------
-# build common libraries by platforms
 include(../CommonCompilerOptions.CMake)
 
-# ------------------------------------------
-# Set OSX specific runtime options∑
-# TODO: use single architecture to avoid Crypto3 error - removed ";arm64"  
-set(CMAKE_OSX_ARCHITECTURES "x86_64")
-set(CMAKE_MACOSX_RPATH 1)
-set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM "BOTH")
-
-if (${CMAKE_BUILD_TYPE} STREQUAL "Debug")
-	set(BUILD_VARIANT "--debug")
-else()
-	set(BUILD_VARIANT "")
-endif(${CMAKE_BUILD_TYPE} STREQUAL "Debug")
-
-# Assuming that xcode 12.0 is installed you most probably have ios sdk 14.0 or later installed (tested on Big Sur)
-# if you don't set a deployment target it will be set the way you only get 64-bit builds
-if(NOT DEFINED DEPLOYMENT_TARGET AND XCODE_VERSION_INT VERSION_GREATER 12.0)
-  # Temporarily fix the arm64 issues in CMake install-combined by excluding arm64 for simulator builds (needed for Apple Silicon...)
-  set(CMAKE_XCODE_ATTRIBUTE_EXCLUDED_ARCHS[sdk=iphonesimulator*] "arm64")
-endif()
-
-set(_CMAKE_COMMON_BUILD_PARAMS ${_CMAKE_COMMON_BUILD_PARAMS}
-   	-DCMAKE_SYSTEM_NAME:STRING=${CMAKE_SYSTEM_NAME}
-    -DCMAKE_OSX_ARCHITECTURES:STRING=${CMAKE_OSX_ARCHITECTURES}
-    -DCMAKE_XCODE_ATTRIBUTE_ONLY_ACTIVE_ARCH:BOOL=YES
-    -DCMAKE_IOS_INSTALL_COMBINED:BOOL=TRUE
-    -DCMAKE_XCODE_ATTRIBUTE_VALID_ARCHS:STRING=${CMAKE_OSX_ARCHITECTURES}
-    -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=${CMAKE_OSX_DEPLOYMENT_TARGET}
-    -D"CMAKE_XCODE_ATTRIBUTE_EXCLUDED_ARCHS[sdk=iphonesimulator*]":STRING=arm64
+# OSX specific runtime options
+set(_CMAKE_COMMON_CACHE_ARGS ${_CMAKE_COMMON_CACHE_ARGS}
+    -DPLATFORM:STRING=${PLATFORM}
 )
```

Comment:
We define that above (`set(PLATFORM MAC CACHE STRING "Toolchain platform parameter")`), but this is not ideal since users with `arm` chips will cross compile to `x86`. I think we can do the following:

```cmake
if(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
    set(_PLATFORM "MAC")
elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "arm")
    set(_PLATFORM "MAC_ARM64")
endif()

set(PLATFORM _PLATFORM CACHE STRING "Toolchain platform parameter")
```

Therefore, if the user wants to cross compile, he can just use `cmake ... -DPLATFORM=Foobar` and the cache entry will be correctly set, otherwise we provide a better default.

Reply:
Perfect, we just might need to update the readme if we decide to hardcode an arch